extensions/core/deployment/src/test/java/io/quarkiverse/cxf/deployment/test/UberJarMergedResourcesTest.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkiverse.cxf.deployment.test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.logging.LogRecord;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.runtime.annotations.QuarkusMain;
+import io.quarkus.test.ProdBuildResults;
+import io.quarkus.test.ProdModeTestResults;
+import io.quarkus.test.QuarkusProdModeTest;
+
+public class UberJarMergedResourcesTest {
+
+    @RegisterExtension
+    static final QuarkusProdModeTest runner = new QuarkusProdModeTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClass(UberJarMain.class))
+            .setApplicationName("uber-jar")
+            .setApplicationVersion("0.1-SNAPSHOT")
+            .setRun(true)
+            .setExpectExit(true)
+            .overrideConfigKey("quarkus.package.type", "uber-jar")
```

Comment:
I suspect this is not doing what we'd hope for. These `@RegisterExtension` tests IIRC always work with an in-memory class-loader and I think `.overrideConfigKey("quarkus.package.type", "uber-jar")` has no effect. This can be proven by a simple experiment: when all changes in this PR except the test are reverted, the test should fail, but in reality it does not (I checked). I think we need a proper integration test. 

For reading the log from an integration test, you need two things:

1. `quarkus.log.file.enable = true` in [application properties](https://github.com/apache/camel-quarkus/blob/main/integration-test-groups/foundation/direct/src/main/resources/application.properties) 
2. A piece of Awaitility code repeatedly reading the whole log file: https://github.com/apache/camel-quarkus/blob/main/integration-test-groups/foundation/direct/src/test/java/org/apache/camel/quarkus/component/direct/it/DirectTest.java#L63-L66

Reply:
@ppalaga Reverting the PR (except for the unit test) failed the test for me when I had tried it locally?  Not sure why you're seeing something different - but I'll look into it later this week and keep you posted.

Reply:
Yes, thanks for explaining, indeed it is failing as expected with those log messages modified.